actionview/lib/action_view/template.rb

```diff
@@ -7,7 +7,7 @@ module ActionView
   class Template
     extend ActiveSupport::Autoload
 
-    STRICT_LOCALS_REGEX = /\#\s+locals:\s+\((.*)\)/
+    STRICT_LOCALS_REGEX = /\#\s+locals:\s+\((.*?)\)(?=\s*-?%>)/m
```

Comment:
So this works for ERB, but I fear that some people were using it with HAML or Slim :/